ros/src/tl_detector/tl_detector.py

```diff
@@ -203,41 +206,59 @@ def process_traffic_lights(self):
 
         """
         light = None
+        car_position = 0
 
         # List of positions that correspond to the line to stop in front of for a given intersection
         #stop_line_positions = self.config['stop_line_positions']
 	
         ntl_wp = -1
         gt_ntl_state =TrafficLight.UNKNOWN
-	ntl_state =TrafficLight.UNKNOWN
+        ntl_state =TrafficLight.UNKNOWN
+
+        if self.light_classifier is not None: 
+            if not self.busy:
+                self.busy = True
+                ntl_state = self.get_light_state(light)
+                self.previous_light_state = ntl_state
+                self.busy = False
+            else:
+                ntl_state = self.previous_light_state
+
 
         if(self.pose):
             car_position = self.get_closest_waypoint(self.pose.pose.position.x,self.pose.pose.position.y, self.waypoints)
-        if car_position:
-            for tl in self.lights: 	
-                nearest_waypoint = self.get_closest_waypoint(tl.pose.pose.position.x,tl.pose.pose.position.y, self.waypoints)
-                if nearest_waypoint > car_position:
-                    ntl_wp = nearest_waypoint
-                    gt_ntl_state = tl.state
-                    break
+
+        # State = 0 : Red
+        if ntl_state != 4:
+            if car_position:
+                for tl in self.lights: 	
+                    nearest_waypoint = self.get_closest_waypoint(tl.pose.pose.position.x,tl.pose.pose.position.y, self.waypoints)
+                    if nearest_waypoint > car_position:
+                        ntl_wp = nearest_waypoint
+                        gt_ntl_state = tl.state
```

Comment:
I would like to keep the functionality of being able to disable the classifier to use the ground truth pass through by manually setting `self.light_classifier = None`, but if I do this, the `ntl_state` will not be updated so the `gt_ntl_state` will also not be updated.  Can we remove the `if ntl_state != 4:` condition on this loop or change it to `if ntl_state != 4 or not self.light_classifier:`?

Reply:
Good catch !